var/spack/repos/builtin/packages/relax/package.py

```diff
@@ -0,0 +1,61 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Relax(CMakePackage):
+    """A set of Reflex libraries for the most common used general data types in
+       the LHC Computing Grid"""
+
+    homepage = "https://twiki.cern.ch/twiki/bin/view/LCG/RELAX"
+    url      = "http://lcgpackages.web.cern.ch/lcgpackages/tarFiles/sources/RELAX-1.tar.gz"
+
+    version('root6', sha256='48ffdddfa0a392fc8ecb9ac17513796c5342acd84b9b11a79c75debc84e6537a')
+
+    depends_on('clhep')
+    depends_on('gsl')
+    depends_on('hepmc@:2.99.99')
+    depends_on('root@6.0.0:')
+
+    # FIXME: Check if this is the best way on the spack side of things
+    def flag_handler(self, name, flags):
+        if name in ['cxxflags', 'cppflags']:
+            # cxxstd in use must match ROOT's
+            cxxstd = self.spec['root'].variants['cxxstd'].value
+            flags.append("-std=c++{}".format(cxxstd))
```

Comment:
So, the intent here is to say that RELAX must be built against the same C++ standard as its ROOT dependency is. I don't like the ways this is currently done for a few reasons:

- `flag_handler` feels like a heavyweight tool for doing this.
- I don't like matching against a hardcoded set of C++ standard flags
- `-std=c++{}` is not 100% portable across compilers.

Do you have any suggestion of more elegant ways of doing this?

Reply:
There are the `cxx98_flag` and friends methods of the compiler class

Reply:
That requires an indefinitely growing switch statement, which I am not terribly fond of. Should I contribute an alternate interface which, given a C++ standard version, provides the right flag?

Reply:
An alternate possibility would be to take advantage of the fact that this is a CMake package and try to play with CMAKE_CXX_STANDARD. While still imperfect, it does look like it could address most of my issues above, and might work with another package I'm now working on which doesn't like the above trick.

Reply:
Works well enough in my experience, so I switched to this for now.